docs/designs/0000-template.md

```diff
@@ -0,0 +1,23 @@
+# Title
+
+Updated by: [0000-this-doc-does-not-exist]()
+
+Obsoleted by: [0000-this-doc-does-not-exist]()
+
+*(these sections should only be added when necessary, and should always include the links to the relevant documents)*
+
+## Introduction
+
+This section should bring a short summary of what this document is covering, the main points it is going to describe, and why it is important. Ideally should be kept under two paragraphs.
+
+## Context [optional]
+
+Explains the context outside the mere technical aspects of the design that existed when it was written. It must make clear what other influences where at play that affected the decisions that were taken. In case there is nothing relevant to mention about the context, this section can be skipped.
+
+## [Free-form sections]
+
+The actual content of the design should be placed under one or more sections that can be named freely. Note that they should be level 2 headings (H2), but each section can also be subdivided by nested headings (H3, H4, H5 or H6) if needed. Only a single level 1 heading (H1) should exist, and it is strictly reserved for the title.
+
+## Decision/Outcome [pick one]
+
+Document here what decision was made, or what outcome is expected out of the design that was created. In case several options were presented in the design, this section must clearly state which one was chosen and the reasons for it. In case of a new package manager design, or a completely new feature to an existing package manager, this section can only include a brief summary of what the implementation will cover, since it is implied that the decision.
```

Comment:
nitpick: Is the description of the intended content for these sections being purposefully done so that it can be rendered online? I mean, the only thing I expect from a template is to be copy-pasted, renamed, filled out and proposed. So, if we make these description commentaries, people can even keep those comments in their proposed doc and that information won't be rendered, just a  bit of thinking out loud.